services/orchest-webserver/client/src/contexts/GlobalContext.tsx

```diff
@@ -0,0 +1,433 @@
+import { useAsync } from "@/hooks/useAsync";
+import {
+  OrchestConfig,
+  OrchestServerConfig,
+  OrchestUserConfig,
+  ReducerActionWithCallback,
+} from "@/types";
+import { ButtonProps } from "@mui/material/Button";
+import { fetcher, hasValue } from "@orchest/lib-utils";
+import React from "react";
+
+type PlainTextToHtmlProps = { text: string };
+const PlainTextToHtml = ({ text }: PlainTextToHtmlProps) => {
+  if (text === undefined) return null;
+  const linesArr = text.split("\n");
+  return (
+    <>
+      {linesArr.map((line, index) => {
+        if (index !== linesArr.length - 1) {
+          return (
+            <>
+              {line}
+              <br />
+            </>
+          );
+        } else {
+          return line;
+        }
+      })}
```

Comment:
Don't forget those keys!

```suggestion
      {linesArr.map((line, index) => {
        if (index !== linesArr.length - 1) {
          return (
            <React.Fragment key={index}>
              {line}
              <br />
            </>
          );
        } else {
          return <React.Fragment key={index}>line</React.Fragment>;
        }
      })}
```

Reply:
AFAIK, Fragments don't need a key, as they are not actually rendered. It also didn't throw warning without the keys. Could you verify?

Reply:
So—I'll have a look to verify, but AFAIK React will always throw an error for anything that doesn't have a key when you render an array, regardless if you have a `div` or `React.Fragment`. This is because ReactDOM will not be able to calculate what has to be re-rendered unless these keys are included—it's a performance optimization more than anything.

IMHO keys should always be included when rendering an array, it might just be pragmatism, I'll try to verify. :+1:

Reply:
Added.

Reply:
Just to verify, here's a minimal example using `React.Fragment`: 

https://codesandbox.io/s/intelligent-surf-8wwmkt?file=/src/App.js